geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/PRColocationDUnitTest.java

```diff
@@ -2604,6 +2612,11 @@ public void rebalancingOrRecoveryFinished(Region region) {
     public void waitForRegion(Region region, long timeout) throws InterruptedException {
       long start = System.currentTimeMillis();
       synchronized (this) {
+        long waitStart = (timeout / 10) - (System.currentTimeMillis() - start);
+        this.wait(waitStart);
+        if (!recoveryStartedOnRegions.contains(region)) {
```

Comment:
I don't know the answer to this, but is there something we could wait for rather than just a timeout?